atlasdb-commons/src/main/java/com/palantir/common/base/Throwables.java

```diff
@@ -79,6 +80,22 @@ public static RuntimeException rewrapAndThrowUncheckedException(String newMessag
         throw createPalantirRuntimeException(newMessage, ex);
     }
 
+    /**
+     * If the provided Throwable is
+     *   a) an ExecutionException or InvocationTargetException, then apply this method on the cause;
+     *   b) a RateLimitExceededException or an AtlasDbDependencyException, then rethrow it;
+     *   c) none of the above, then wrap it in an AtlasDbDependencyException and throw that.
+     */
+    public static RuntimeException unwrapAndThrowRateLimitExceededOrAtlasDbDependencyException(Throwable ex) {
```

Comment:
should we just replace `unwrapAndThrowAtlasDbDependencyException` with this method?

Reply:
is there ever a case where we'd want to use that one and not this one?

Reply:
Yeah, for TimeLock (which isn't QoS serviced I think).